samplelistener/listener/dispatcher_service.bal

```diff
@@ -2,46 +2,45 @@ import ballerina/http;
 import ballerina/jballerina.java;
 
 service class DispatcherService {
-   private SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef;
-   //private string verificationToken;
+     private map<GenericService> services;
 
-   isolated function init(SlackAppMentionHandlingService|SlackAppCreatedHandlingService serviceRef) returns error? { //Receive verification token
-        self.serviceRef = serviceRef;
-        //self.verificationToken = "9asdbas9009123nas1e2";
+   isolated function init() returns error? {
+        self.services = {};
    }
 
-   isolated resource function post events (http:Caller caller, http:Request request) returns error? {
-        json payload = check request.getJsonPayload();
-        // string eventOrVerification = check payload.'type;
-
-        // if (payload.token !== self.verificationToken) {
-        //     return error("Verification token mismatch");
-        // }
+   isolated function addServiceRef(string serviceType, GenericService genericService) {
+        self.services[serviceType] = genericService;
+   }
 
-        // if (eventOrVerification == URL_VERIFICATION) {
-        //     check self.verifyURL(caller, payload);
-        // } else if (eventOrVerification == EVENT_CALLBACK) {
-        // }
+   resource function post events (http:Caller caller, http:Request request) returns error? {
+        json payload = check request.getJsonPayload();
```

Comment:
We are doing this instead of using data-binding because of a bug, right? Shall we add a comment mentioning it?

Reply:
Resolved